TrkDiag/src/TrkMCTools.cc

```diff
@@ -85,17 +85,22 @@ namespace mu2e {
         // loop over the hits and find the associated steppoints
         bool isactive = tshs.flag().hasAllProperties(active);
         StrawDigiMC const& mcdigi = mcdigis.at(tshs.index());
-        art::Ptr<SimParticle> spp = mcdigi.earlyStrawGasStep()->simParticle();
-        // see if this particle has already been found; if so, increment, if not, add it
-        bool found(false);
-        for(auto& spc : sct ) {
-          if(spc._spp == spp ){
-            found = true;
-            spc.append(spp,isactive);
-            break;
+        // if mc info is not meaningful, then skip this digi.
+        // this looks sketchy, but nowhere is an implicit association
+        // between the sct and mcdigis collection actually assumed
+        if (mcdigi.provenance() != DigiProvenance::External){
```

Comment:
this logic would fail to do what I think is intended if provenance() == Unknown.
I suggest adding a 'bool hasSim()' member to DigiProvenance, returning true if provenance == Sim || provenance == Mixed.  Then this line (and similar lines in other codes) becomes: if(mcdigi.provenance().hasSim()){....

Reply:
This requires inheriting from an `EnumToStringSparse<>` specialization, which is more delicate than I'd have guessed (due to `explicit` constructors and the details of assignment operators). @kutschke may want to validate what I came up in my last push --- it defines one extra constructor and sprinkles is in some intermediary `static_cast<>`s. This allows us to construct the subclass and have the functionality you requested, but at the cost of only reimplementing one simple constructor.

Reply:
Thanks for giving me the heads up on this - see comments in my most recent review.